lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced-tabbed.component.ts

```diff
@@ -0,0 +1,122 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component } from '@angular/core';
+import { Subject } from 'rxjs';
+import { SearchDateRangeAdvanced } from './search-date-range-advanced/search-date-range-advanced';
+import { DateRangeType } from './search-date-range-advanced/date-range-type';
+import { SearchWidget } from '../../models/search-widget.interface';
+import { SearchWidgetSettings } from '../../models/search-widget-settings.interface';
+import { SearchQueryBuilderService } from '../../services/search-query-builder.service';
+import { InLastDateType } from './search-date-range-advanced/in-last-date-type';
+
+@Component({
+  selector: 'adf-search-date-range-advanced-tabbed',
+  templateUrl: './search-date-range-advanced-tabbed.component.html',
+  styleUrls: ['./search-date-range-advanced-tabbed.component.scss']
+})
+export class SearchDateRangeAdvancedTabbedComponent implements SearchWidget {
+    displayValue$ = new Subject<string>();
+    id: string;
+    startValue: SearchDateRangeAdvanced = {
+        dateRangeType: DateRangeType.ANY,
+        inLastValueType: InLastDateType.DAYS,
+        inLastValue: undefined,
+        betweenStartDate: undefined,
+        betweenEndDate: undefined
+    };
+    settings?: SearchWidgetSettings;
+    context?: SearchQueryBuilderService;
+    fields: string[];
+    displayedLabelsByField: { [key: string]: string } = {};
+    queries: { [key: string]: string } = {};
+    valuesToDisplay: { [key: string]: string } = {};
+    tabsValidity: { [key: string]: boolean } = {};
+    combinedQuery: string;
+    combinedValuesToDisplay: string;
+
+    private value: { [key: string]: Partial<SearchDateRangeAdvanced> } = {};
+
+    getCurrentValue(): { [key: string]: Partial<SearchDateRangeAdvanced> } {
+        return this.value;
+    }
+
+    hasValidValue(): boolean {
+        return !Object.values(this.tabsValidity).some((valid) => !valid);
```

Comment:
This looks like a bit weird negation, I think it would be more readable if you use `.every` and check if each is valid instead of checking if some is invalid and negating it

Reply:
Updated the code to use `.every()` instead. This looks much better now, thanks!!